pkg/kubelet/pod_workers.go

```diff
@@ -217,33 +217,25 @@ func (p *podWorkers) UpdatePod(options *UpdatePodOptions) {
 		// comment in syncPod.
 		go func() {
 			defer runtime.HandleCrash()
-			p.managePodLoop(podUpdates)
+			p.managePodLoop(uid, podUpdates)
 		}()
 	}
-	if !p.isWorking[pod.UID] {
-		p.isWorking[pod.UID] = true
-		podUpdates <- *options
-	} else {
-		// if a request to kill a pod is pending, we do not let anything overwrite that request.
-		update, found := p.lastUndeliveredWorkUpdate[pod.UID]
-		if !found || update.UpdateType != kubetypes.SyncPodKill {
```

Comment:
Does removing this make it possible to lose pod kill requests?

Reply:
Maybe, when `podUpdate` is full, it will lose the kill request.

Reply:
I'd prefer that we have a way to ensure kill requests are not lost, as it was clearly important enough to take special precautions when the original code was written.
@yujuhong thoughts?